pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -785,6 +786,18 @@ public boolean isBackwardCompatibleWith(Schema oldSchema) {
         return false;
       }
     }
+
+    List<String> primaryKeyColumns = getPrimaryKeyColumns();
+    List<String> oldPrimaryKeyColumns = oldSchema.getPrimaryKeyColumns();
+
+    if (primaryKeyColumns == null) {
+      return oldPrimaryKeyColumns == null;
+    }
+
+    if (!Arrays.equals(primaryKeyColumns.toArray(), oldPrimaryKeyColumns.toArray())) {
+      return false;
+    }
+
     return true;
```

Comment:
can be replaced with `return Arrays.equals(primaryKeyColumns.toArray(), oldPrimaryKeyColumns.toArray())`?

Reply:
no, i left it this way so that other folks can add new conditions in the future in same pattern.

Reply:
Got it. We can atleast avoid the null check block above as `Arrays.equals` will handle that check internally.

Reply:
actually, there was this `// TODO: Add the reason of the incompatibility`

would it be ok to fix this TODO in this PR as well, since we're adding more checks for incompatibility.